codec-http2/src/main/java/io/netty/handler/codec/http2/UniformStreamByteDistributor.java

```diff
@@ -88,33 +87,23 @@ public void updateStreamableBytes(StreamState streamState) {
     public boolean distribute(int maxBytes, Writer writer) throws Http2Exception {
         checkNotNull(writer, "writer");
 
-        // First, write out any empty frames.
-        if (emptyFrameQueue != null) {
-            while (!emptyFrameQueue.isEmpty()) {
-                State state = emptyFrameQueue.remove();
-                state.enqueued = false;
-                if (state.streamableBytes > 0) {
-                    // Bytes have been added since it was queued. Add it to the regular queue.
-                    state.addToQueue();
-                } else {
-                    // Still an empty frame, just write it.
-                    state.write(0, writer);
-                }
-            }
-        }
-
         final int size = queue.size();
-        if (size == 0 || maxBytes <= 0) {
+        if (size == 0) {
             return totalStreamableBytes > 0;
         }
 
         final int chunkSize = max(minAllocationChunk, maxBytes / size);
 
-        // Write until the queue is empty or we've exhausted maxBytes. We need to check queue.isEmpty()
-        // here since the Writer could call updateStreamableBytes, which may alter the queue.
-        do {
-            // Remove the head of the queue.
-            State state = queue.remove();
+        State state;
+        while ((state = queue.peek()) != null) {
```

Comment:
we know on the first iteration the peek() will not return null because we previously checked size. Consider refactoring:

```java
State state = queue.pollFirst();
do {
  state.enqueued = false;
  if (state.streamableBytes > 0 && maxBytes == 0) {
    // Stop at the first state that can't send. Note that empty frames at the head
    // of the queue will always get written.
    break;
  }
...
} while ((state = queue.pollFirst()) != null);